astrolabe/validator.py

```diff
@@ -73,19 +71,8 @@ def run_test(self, driver_workload):
 
         try:
             stats = subprocess.stop()
-        except TimeoutExpired:
-            self.fail("The workload executor did not terminate soon after "
-                      "receiving the termination signal.")
-
-        # Check that results.json is actually written.
-        if all(val == -1 for val in stats.values()):
-            self.fail("The workload executor did not write a results.json "
-                      "file in the expected location, or the file that was "
-                      "written contained malformed JSON.")
-
-        if any(val < 0 for val in stats.values()):
-            self.fail("The workload executor reported incorrect execution "
-                      "statistics. Reported statistics MUST NOT be negative.")
+        except WorkloadExecutorError as exc:
+            self.fail("WorkloadExecutorError: %s" % exc)
```

Comment:
`DriverWorkloadSubprocessRunner.stop()` actually never raised a `TimeoutExpired` error. It was caught internally and re-thrown as a `WorkloadExecutorError`.